Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -117,6 +116,282 @@ class UpstreamIntegrationTests: TestBase {
+    }
+
+    // Tests standard sendEvent with complex XDM - many keys and different value types
+    func testSendEvent_withEventXDMOnly_receivesExpectedEventHandles() {
+        // Setup
+        let standardNetworkRequest = NetworkRequest(urlString: "https://obumobile5.data.adobedc.net/ee/v1/interact", httpMethod: .post)!
+        networkService.setExpectationForNetworkRequest(networkRequest: standardNetworkRequest, expectedCount: 1)
+
+        let eventPayloadJSON = #"""
+        {
+          "xdm": {
+            "testString": "xdm",
+            "testInt": 10,
+            "testBool": false,
+            "testDouble": 12.89,
+            "testArray": ["arrayElem1", 2, true],
+            "testDictionary": {
+              "key": "val"
+            }
+          }
+        }
+        """#
+
+        let xdm = getAnyCodableAndPayload(eventPayloadJSON, type: .xdm)!
+
+        let experienceEvent = ExperienceEvent(xdm: xdm.payload)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // MARK: Network response assertions
+        let matchedResponse = networkService.getResponsesFor(networkRequest: standardNetworkRequest, timeout: 5)
+        XCTAssertEqual(200, matchedResponse.first?.responseCode)
+
+        // MARK: Response Event assertions
+        // Only validate for the location hint relevant to Edge Network extension
+        let expectedLocationHintJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedLocationHintJSON, eventSource: TestConstants.EventSource.LOCATION_HINT_RESULT, exactMatchPaths: ["payload[*].scope"])
+
+        let expectedStateStore1stJSON = #"""
+        {
+          "payload": [
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            },
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedStateStore1stJSON, eventSource: TestConstants.EventSource.STATE_STORE)
+    }
+
+    // MARK: - Configuration tests
+    // Tests standard sendEvent with both XDM and data, where data is complex - many keys and
+    // different value types
+    func testSendEvent_withSetLocationHint_receivesExpectedEventHandles() {
+        // Setup
+        let locationHintNetworkRequest = NetworkRequest(urlString: "https://obumobile5.data.adobedc.net/ee/va6/v1/interact", httpMethod: .post)!
+        networkService.setExpectationForNetworkRequest(networkRequest: locationHintNetworkRequest, expectedCount: 1)
+
+        Edge.setLocationHint("va6")
```

Comment:
can we have a constant inside this func for actual location hint set, and you can reuse it for the expectations?